### PR TITLE
add a script to create a debug tarball

### PR DESCRIPTION
**Pull request by qrkourier**

### CODE REVIEW

**Comment by qrkourier** [comment]
Should I filter out any of those system calls in the *.strace file, or are they all potentially useful?

**Comment by qrkourier** [comment]
```bash
 ❯ sudo bash ./scripts/debug.bash         
INFO: debug bundle created at /tmp/ziti-edge-tunnel-0.22.20-local-2024-01-04T21:54Z.tgz from files in /tmp/tmp.COFIjI3FXE
```

File: scripts/debug.bash [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    ) &> "${HOST_INFO_FILE}"
+    echo -n "."
+    
+    # save the current service unit invocation's log messages
+    journalctl _SYSTEMD_INVOCATION_ID="$(systemctl show -p InvocationID --value ziti-edge-tunnel.service)" -l --no-pager \
+    &> "${LOG_FILE}"
+    echo -n "."
+    
+    # save the threads and backtrace
+    timeout --kill-after=1s 3s \
```
Comment (scareything) [inline scripts/debug.bash:80]: `backtrace` is only going to show the stack for a specific thread, and probably not the most interesting one. we can get a backtrace for all threads with this gdb command:
```
thread apply all backtrace
```

File: scripts/debug.bash [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        exit 1
+    fi
+    
+    # save the systemd-resolved state
+    (
+        set +e
+        set -x
+        systemctl status systemd-resolved.service
+        resolvectl status
+        resolvectl dns ziti0
```
Comment (scareything) [inline scripts/debug.bash:57]: fyi the tun device might not be `ziti0`

**Comment by scareything** [comment]
It's be good to get a few more tidbits, like `ip link show`, `ip addr show`, `ip route show`

File: scripts/debug.bash [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    # save the threads and backtrace
+    timeout --kill-after=1s 3s \
+        gdb /opt/openziti/bin/ziti-edge-tunnel \
+            --pid "${ZET_PID}" \
+            --batch \
+            --ex "set verbose on" \
+            --ex "set pagination off" \
+            --ex "info threads" \
+            --ex "backtrace" \
+            --ex "quit" \
```
Comment (scareything) [inline scripts/debug.bash:88]: I'd expect the `quit` command to cause a prompt. Does it? Either way it's probably safest to `--ex "detach"` first.

Comment (qrkourier) [reply scripts/debug.bash:88]: Correct, `--batch` was necessary to avoid a hung `gdb` command that attaches.

File: scripts/debug.bash (hunk 2 above)
Comment (qrkourier) [reply scripts/debug.bash:57]: I'll stop specifying the device for both of those commands.

**Comment by qrkourier** [comment]
@scareything Thanks for those refinements. It's ready for another review.

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**